### ISSUE TITLE
Impact of Byte Queue Limit on AVTP

### ISSUE THREAD

**Issue by raosush** [Avnu/libavtp#40]
Hi,
I have been reading the IEEE1722 document and have also been experimenting with this implementation of AVTP.
I had also been going through the TC components of the Linux kernel and came across [Byte Queue Limit(BQL)](https://tldp.org/en/Traffic-Control-HOWTO/ar01s04.html)(Section 4.10).
I stumbled upon the following thought after reading about BQL:
- **Would BQL impact the performance of AVTP?** 
  - BQL aims to improve the utilisation of the queue by appropriately queuing certain amount of data. AVTP, built on top of TSN(which requires ultra low latency), requires that the latency be bounded and be as low as possible. With BQL, the queuing delay can significantly be reduced as it tries to maintain the queues at the most optimal size, in an attempt to increase the utilisation of the link.

I request all of you to share your thoughts regarding the same.

Thank you in advance!

**Comment by edersondisouza** [comment]
@vcgomes ^

**Comment by vcgomes** [comment]
Yes, in theory, BQL will add a bit of overhead to the transmission path, but from my mental model, the impact should be minor when the transmissions are well behaved, i.e. less or equal to the configured rates (when using CBS, for example). 

In the case of less well behaved transmissions you should see increased latency because the packets are being held for longer periods of time, because the transmission queue will be "busy" for longer.

Anyway, it might be worth to disable CONFIG_BQL in your kernel config and do some experiments.
